coralproject/talk
# [CORL-1655] Show trace ID's to resolve client error's easier

## What does this PR do?

Creates a unique trace ID on every server request that can be attached to error messages, logged in the server logs, and shown to the user. The trace ID is oblique in that it tells the user nothing of what caused the error, but is an identifier they can provide to support staff that can easily be correlated to server logs to assist in resolving/debugging the issue.

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## How do I test this PR?

- Throw an error somewhere in the code base
  - An easy one is to make the `body` resolver throw an error in `src/core/server/graph/resolvers/Comment.ts` (Ln: 56)
- Visit the site and trigger this error
- Should see a call out with the trace ID
- Check console running Coral and find matching `contextID` for the error
- Test this in both development and production builds

Example:
<img width="462" alt="image" src="https://user-images.githubusercontent.com/5751504/122956375-2e1cbb00-d33e-11eb-8291-4a3600401d50.png">
